.github/workflows/actions.yml

```
@@ -0,0 +1,22 @@
+on:
+  # Trigger the workflow on push or pull request,
+  # but only for the main branch
+  push:
+    branches:
+      - main
+      - github-actions
+  pull_request:
+    branches:
+      - main
```

Here, I suggest to trigger for every push.
Why? If we push new code to a branch, a build is triggered. The person is getting quick feedback.
Even if the person opens a PR, the build status is reported.

Additionally, we compile it every night. Why? To check for early regressions in 3rd party packages.
```suggestion
  push:
  schedule:
    - cron: "5 1 * * *"
```